pubsub/index.js

```diff
@@ -114,28 +124,25 @@ async function main () {
   })
 
   subscription.on('error', error => {
-    console.log(timestamp(), error.stack)
+    logger.error({ type: 'subscription.error', error }, 'Subscription error')
   })
 
   subscription.on('close', () => {
-    console.log(timestamp(), 'Error: subscription closed')
+    logger.fatal({ type: 'subscription.close' }, 'Subscription closed')
     process.exit(1)
   })
 }
 
-function timestamp () {
-  return new Date().toISOString()
-}
-
 function setupCargo (endpoint, key) {
   const cargo = async.cargo(async payload => {
     try {
       await sendPayload(payload, endpoint, key)
       clearMessages(payload, message => message.ack())
-      console.log(timestamp(), 'Success!', endpoint, payload.length)
+      logger.info({ type: 'events.processed', endpoint, count: payload.length }, 'Events processed')
     } catch (error) {
-      console.log(timestamp(), endpoint, error.stack)
-      clearMessages(payload, message => message.nack(), true)
+      logger.error({ type: 'events.error', endpoint, error, count: payload.length }, 'Events error')
+      // Smear nacks over 5 minute period
+      clearMessages(payload, message => message.nack(60 + (Math.random() * 240)), true)
```

Comment:
Not sure this does anything, as the GCP docs don't mention supporting a delay argument to `Message.nack()`: https://googleapis.dev/nodejs/pubsub/latest/Message.html#nack